src/ui/hook/useDropDirection.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect, useCallback, type RefObject } from 'react'
+
+export const useDropDirection = (ref: RefObject<HTMLElement>): boolean => {
+  const [dropup, setDropup] = useState(false)
+
+  const updateDropDirection = useCallback(() => {
+    if (ref.current) {
+      setDropup(ref.current.getBoundingClientRect().y > window.innerHeight / 2)
+    }
+  }, [ref])
+
+  useEffect(() => {
+    updateDropDirection()
+
+    window.addEventListener('resize', updateDropDirection)
```

Comment:
Yes, to embrace SRP, I would have split this hook into 2 hooks: `useWindowResize` and `useDropDirection` because the first one could be invoked for another usage and should not depend on the dropdown implementation.

By the way, I would add that if you create a `useWindowResize` hook, I'm not sure that useDropDirection remains relevant as it could be implemented straight into the component (if never used outside of the dropdown itself), IMHO

Reply:
In fact, the hook becomes more meaningful when we extend its generalization slightly. This is because the hook currently encapsulates the evaluation function, but it could be abstracted to allow for behavior customization. For instance, we could parameterize the behavior to determine bottom positioning based on available window space (as it is currently) or to identify an element's viewport membership, among other possibilities... Of course this goes beyond what we need right now, but it's just to give a little perspective...